datapackage/registry.py

```diff
@@ -107,9 +108,20 @@ def _get_profile(self, profile_id):
 
     def _get_registry(self, registry_path_or_url):
         '''dict: Return the registry as dict with profiles keyed by id.'''
-        table = tabulator.Stream(registry_path_or_url, headers=1).open()
+        if registry_path_or_url.startswith('http'):
+            profiles = self._load_json_url(registry_path_or_url)
+        else:
+            profiles = self._load_json_file(registry_path_or_url)
         try:
-            registry = dict([(o['id'], o) for o in table.read(keyed=True)])
+            registry = {}
+            for profile in profiles:
```

Comment:
Or, are you doing this handling of profiles for backwards compatibility? If yes, use warnings.